### PR TITLE
Add compiler-rt

### PR DESCRIPTION
**Pull request by jakirkham**
Adds a recipe for LLVM's `compiler-rt`.

cc @isuruf

### CODE REVIEW

**Comment by jakirkham** [comment]
Thoughts @conda-forge/staged-recipes?

File: recipes/compiler-rt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  license_file: LICENSE.TXT
+  summary: C++ standard library
+
+extra:
+  recipe-maintainers:
+    - SylvainCorlay
+    - inducer
+    - mattwala
+    - jakirkham
+    - isuruf
```
Comment (scopatz) [inline recipes/compiler-rt/meta.yaml]: @SylvainCorlay @inducer @mattwala @isuruf - are you all OK with being maintainers?

Comment (SylvainCorlay) [reply recipes/compiler-rt/meta.yaml]: sure

Comment (jakirkham) [reply recipes/compiler-rt/meta.yaml]: @inducer @mattwala @isuruf, thoughts?

Comment (isuruf) [reply recipes/compiler-rt/meta.yaml]: sure

Comment (scopatz) [reply recipes/compiler-rt/meta.yaml]: pinging this again :)

Comment (scopatz) [reply recipes/compiler-rt/meta.yaml]: @jakirkham - maybe you should drop the folks who haven't explicitly agreed. They can always add themselves in the feedstock :)

Comment (inducer) [reply recipes/compiler-rt/meta.yaml]: Sure, sorry!

Comment (jakirkham) [reply recipes/compiler-rt/meta.yaml]: Ok, dropped @mattwala. Everyone else has ok'd this.

**Pull request merged**